lib/teamcity/oracle/oracle_installation.txt

```diff
@@ -1,12 +1,15 @@
 
-Installation steps for Oracle (Express Edition) for NH TeamCity:
+Installation steps for Oracle for NH TeamCity:
 
-1.  Download Oracle Database 10g Express Edition (OracleXE.exe):  http://www.oracle.com/technetwork/database/express-edition/downloads/102xewinsoft-090667.html;
+1.  Download Oracle Database 12c (12.2):  http://www.oracle.com/technetwork/database/enterprise-edition/downloads/index.html
+    A license key is required.
+    Alternatively, from the same page, you may download the free 11g Express edition (XE), but be aware this edition
+    does not succeed all the NHibernate tests.
         (Note: you'll need to sign-up for a (free) Oracle account to download)
 
 2.  Run the installer ... choose any HTTP port for the listener if asked (it may automatically choose the default 8080 if available);
 3.  Enter 'password' as the password for the SYS and SYSTEM accounts;
-4.  The setup should install Oracle (XE) on the machine;
+4.  The setup should install Oracle on the machine;
 5.  Once the installation is complete, leave the option ticked to open the homepage for configuration.
 
 Creating the NH Schema:
```

Comment:
There is no application express, so need to do with SQL+:
```shell
SQL*Plus: Release XXX Production on YYY
Copyright (c) 1982, 2014, Oracle.  All rights reserved.
Enter user-name: / as SYSDBA
Connected to:
Oracle Database XXX
SQL> create user nhibernate identified by nhibernate;
User created.
SQL> grant dba to nhibernate;
Grant succeeded.
SQL>
```

Reply:
If there is ORA-65096. It means that the DB is installed as a container DB. The container DB wants user names to start with `c##`.  To enable creating users without `c##` need to run `alter session set "_ORACLE_SCRIPT"=true;` to allow creating users without c##.  More info [here](http://www.dba-oracle.com/t_ora_65096_create_user_12c_without_c_prefix.htm).

Reply:
There is still a Database Express (at least with enterprise edition, url https://localhost:5500/em/shell#/security/show_users using an non-trusted certificate) which allows to handle this, but the SQL plus solution seems more widely available, so thanks for the instructions.